internal/webhook/v1alpha1/workspace_webhook.go

```diff
@@ -206,9 +215,19 @@ func (v *WorkspaceCustomValidator) ValidateUpdate(ctx context.Context, oldObj, n
 		}
 	}
 
+	originalOwnershipType := getEffectiveOwnershipType(oldWorkspace.Spec.OwnershipType)
 	newOwnershipType := getEffectiveOwnershipType(newWorkspace.Spec.OwnershipType)
+	workspacelog.Info("Ownership validation check", "originalType", originalOwnershipType, "newType", newOwnershipType, "workspace", newWorkspace.GetName())
```

Comment:
could you check the logs? I'm curious in particular if you already get the `Workspace.Name` from context (and the namespace). If not, add both name and namespace please. If they are there, remove the duplicate and rely on the context please.